R/FilterState.R

```diff
@@ -249,12 +249,21 @@ FilterState <- R6::R6Class( # nolint
             `data-bs-toggle` = "collapse",
             href = paste0("#", ns("body")),
             # header elements
-            if (private$is_locked()) icon("lock") else NULL,
-            if (private$is_fixed()) icon("burst") else NULL,
+            if (private$is_anchored()) {
+              if (private$is_fixed()) {
+                icon("anchor-lock")
+              } else {
+                icon("anchor")
+              }
+            } else if (private$is_fixed()) {
+              icon("lock")
+            } else {
+              NULL
```

Comment:
`else NULL` is not necessary because `if (condition)` returns `NULL` by default if the condition is not met.
```suggestion
```